base/inference.jl

```diff
@@ -3276,44 +3276,6 @@ function replace_getfield!(ast, e::ANY, tupname, vals, sv, i0)
     end
 end
 
-function code_typed(f, types::ANY; optimize=true)
-    types = to_tuple_type(types)
-    code_typed(call, Tuple{isa(f,Type)?Type{f}:typeof(f), types.parameters...}, optimize=optimize)
```

Comment:
One thing that can be done is to have the macros generate an escaped `call` symbol, in order to pass the `call` function from the calling context.

The general problem is a case like `methods(call, Tuple{Any, ...`, where you obviously can't look at the first argument to see what module `call` is from.